### ISSUE TITLE
runtimeVM: containerd-shim-kata-v2 binary can't be load from an arbitrary path

### ISSUE THREAD

**Issue by fidencio** [cri-o/cri-o#4541]
**Description**

kata-deploy "installs" `containerd-shim-kata-v2` binary in `/opt/kata/bin`, which is not part of the PATH. Knowing that, when setting it up, I've added the current snippet to the CRI-O's configuration:
```
[crio.runtime.runtimes.kata-qemu]
  runtime_path = "/opt/kata/bin/containerd-shim-kata-v2"
  runtime_type = "vm"
  runtime_root = "/run/vc"
  privileged_without_host_devices = true
```
However, the pod can't be started due to:
```
Warning  FailedCreatePodSandBox  4s (x3 over 33s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = runtime "/opt/kata/bin/containerd.shim.kata.v2" binary not installed "containerd-shim-kata-v2": file does not exist
```
The CRI-O code responsible for spawning the `containerd-shim-kata-v2` can be found [here](https://github.com/cri-o/cri-o/blob/c75afa5b3515a7ab2f6a6c9517f081a3bae75111/internal/oci/runtime_vm.go#L194-L212) and it'll lead to the following snippet in the [containerd code](https://github.com/containerd/containerd/blob/ccde82da2b46b0a4d4cf24576c2499288594df96/runtime/v2/shim/util.go#L41-L112)
```
...
		if cmdPath, lerr = exec.LookPath(name); lerr != nil {
			if eerr, ok := lerr.(*exec.Error); ok {
				if eerr.Err == exec.ErrNotFound {
					// LookPath only finds current directory matches based on
					// the callers current directory but the caller is not
					// likely in the same directory as the containerd
					// executables. Instead match the calling binaries path
					// (containerd) and see if they are side by side. If so
					// execute the shim found there.
					testPath := filepath.Join(filepath.Dir(self), name)
					if _, serr := os.Stat(testPath); serr == nil {
						cmdPath = testPath
					}
					if cmdPath == "" {
						return nil, errors.Wrapf(os.ErrNotExist, "runtime %q binary not installed %q", runtime, name)
					}
				}
			}
		}
...
```

The problem in the code above is that `exec.LookPath(name)` will search for "containerd-shim-kata-v2" in the directories named by the PATH environment variable, and `/opt/kata/bin` isn't one of those, thus it'll just fail.

I can see an improvement done on `containerd` side to actually take into consideration the full binary path, but I think we should also be able to have this set on CRI-O side, mainly considering we already have ways to set environment variables, such as `default_env` and `conmon_env`.

By the way, I've tried setting `/opt/kata/bin` for those two, but the same results.

**Steps to reproduce the issue:**
1. Install kata-deploy, following https://github.com/kata-containers/kata-containers/tree/main/tools/packaging/kata-deploy#kubernetes-quick-start
2. Update CRI-O's configuration file to work with shimv2 (we already have an issue opened for that: https://github.com/kata-containers/kata-containers/issues/1357)
```
[crio.runtime.runtimes.kata-qemu]
  runtime_path = "/opt/kata/bin/containerd-shim-kata-v2"
  runtime_type = "vm"
  runtime_root = "/run/vc"
  privileged_without_host_devices = true
```
3. Try to start a pod using kata-qemu runtimeClass.

**Describe the results you received:**
The pod creation will fail due to:
```
Warning  FailedCreatePodSandBox  4s (x3 over 33s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = runtime "/opt/kata/bin/containerd.shim.kata.v2" binary not installed "containerd-shim-kata-v2": file does not exist
``` 

**Describe the results you expected:**
Everything works as expected.

**Additional information you deem important (e.g. issue happens only occasionally):**
All additional information has been provided already

**Output of `crio --version`:**

```
crio version 1.20.0-dev
Version:       1.20.0-dev
GitCommit:     c75afa5b3515a7ab2f6a6c9517f081a3bae75111
GitTreeState:  clean
BuildDate:     2021-02-04T11:41:56Z
GoVersion:     go1.14.12
Compiler:      gc
Platform:      linux/amd64
Linkmode:      dynamic
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Tested on a single node kubernetes clusters, deployed on a CentOS 8 VM.

**Comment by fidencio** [comment]
One simple solution for this would be:
* Ensure kata-deploy configures `default_env` accordingly;
* Ensure `os.Setenv()` is called at some point before starting the `containerd-shim-kata-v2` binary, forcing then an update to the `PATH;
  * I'm not much comfortable on explicitly calling `os.Setenv(...)` from [`runtimeVM.startRuntimeDaemon()`](https://github.com/cri-o/cri-o/blob/c75afa5b3515a7ab2f6a6c9517f081a3bae75111/internal/oci/runtime_vm.go#L183) and I'd like to hear some input from more experienced developers.

@haircommander, @saschagrunert, @fgiudici, mind to add your input on this?

**Comment by fgiudici** [comment]
I guess enforcing the binary to be in the PATH is for security reasons... but that doesn't make sense when we have a full path specified IMO.
I cannot think of any good reason the ignore a full binary path when specified.

**Comment by fidencio** [comment]
I've opened the following PR for containerd: https://github.com/containerd/containerd/pull/5007
If that gets merged, we could simple update the vendored code.

**Comment by fidencio** [comment]
This can be closed as the PR on containerd side was merged and we already vendor the current code, which includes the fix.
